backend/shared/shared/oauth2/auth.py

```diff
@@ -0,0 +1,129 @@
+import logging
+
+import requests
+from django.contrib.auth.models import Group
+from django.core.exceptions import PermissionDenied
+from django_auth_adfs.backend import AdfsAuthCodeBackend
+from django_auth_adfs.config import provider_config, settings
+from django_auth_adfs.exceptions import MFARequired
+
+LOGGER = logging.getLogger(__name__)
+
+
+class HelsinkiAdfsAuthCodeBackend(AdfsAuthCodeBackend):
+    def get_member_objects_from_graph_api(
+        self, user_oid: str, graph_api_access_token: str
+    ):
+        url = f"https://graph.microsoft.com/v1.0/users/{user_oid}/getMemberObjects"
+
+        headers = {
+            "Authorization": f"Bearer {graph_api_access_token}",
+        }
+
+        data = {"securityEnabledOnly": False}
+
+        response = requests.post(
+            url,
+            json=data,
+            headers=headers,
+        )
+        response.raise_for_status()
+
+        # This will be a list of UUIDs. We will use them as group names in Django.
+        groups = response.json()["value"]
```

Comment:
Could the group name have some kind of indication (prefix or something) that the group was created based on this query? "adfs-{UUID}"?

Also, if the values are supposed to be UUIDS, should it be validated, for example try to convert them to UUID?

Reply:
Good point!

I'll add the prefix there. I have to make sure that they are all UUIDs and I could implement the UUID validation as well.

Reply:
Fixed